### PR TITLE
DEV-5832: add correct_bond_price_quotes()

### PR DESCRIPTION
**Pull request by thomasdotred**
- add test for correct_bond_price_quotes()
- move command line app utility tests to unit test folder

### CODE REVIEW

File: lusidtools/cocoon/utilities.py [hunk 1]
```diff
@@ -730,8 +730,34 @@ def parse_args(args: dict):
...
+
+    price_col = mapping[file_type]["quote_scalar"]["price"]
+    type_col = mapping[file_type]["quote_scalar"]["type"]
+    type_code = mapping[file_type]["quote_scalar"]["type_code"]
+    scale_factor = mapping[file_type]["quote_scalar"]["scale_factor"]
+
+    df["Prices Corrected"] = None
+
+    for index, row in df.iterrows():
+        if row[type_col] == type_code:
```
Comment (rizwansaeed) [inline lusidtools/cocoon/utilities.py]: can make this a ternary assigment:

```
corrected_price = row[price_col] * scale_factor if row[type_col] == type_code else row[price_col]
```

File: tests/unit/cocoon/test_utilities.py [hunk 2]
```diff
@@ -1249,3 +1249,297 @@ def test_checkargs_with_incorrect_type(self, _, function, param):
     def test_checkargs_with_invalid_argument(self, _, function, kwargs):
         with self.assertRaises(ValueError):
             function(**kwargs)
+
+    def test_scale_quote_of_type(self):
```
Comment (rizwansaeed) [inline tests/unit/cocoon/test_utilities.py]: we probably need some more tests here:
- column contains missing/`None` quote values
- no `type_code`s match

File: lusidtools/cocoon/utilities.py [hunk 3]
```diff
@@ -730,8 +730,34 @@ def parse_args(args: dict):
...
+            corrected_price = row[price_col] * scale_factor
+        else:
+            corrected_price = row[price_col]
+        df.at[
+            index, "Prices Corrected"] = corrected_price
```
Comment (rizwansaeed) [inline lusidtools/cocoon/utilities.py]: This isn't really a "correction", perhaps rename to "__adjusted_quote"

If we're adding additional columns we might need to adopt a convention to avoid any potential column name clashes, I'd suggest prefixing with "__" (double underscore)

Comment (rizwansaeed) [reply lusidtools/cocoon/utilities.py]: Actually, we probably need to check if our adjusted column already exists, we have 2 options:
1) raise a ValueError
2) pass in the name for the column and use that

File: tests/unit/cocoon/test_utilities.py (hunk 2 above)
Comment (thomasdotred) [reply tests/unit/cocoon/test_utilities.py]: - add test for None values for not_of type, and of_type
- add test for invalid type & price column names

File: lusidtools/cocoon/utilities.py (hunk 3 above)
Comment (thomasdotred) [reply lusidtools/cocoon/utilities.py]: - changed to `__adjusted_quote`
- ` df["__adjusted_quote"] = None` creates the few a few lines before. Is it really necessary to check it still exists?

File: lusidtools/cocoon/utilities.py (hunk 1 above)
Comment (thomasdotred) [reply lusidtools/cocoon/utilities.py]: changed

File: tests/unit/cocoon/test_utilities.py (hunk 2 above)
Comment (thomasdotred) [reply tests/unit/cocoon/test_utilities.py]: - added test for missing not_of_type values and missing of_type values
- added tests for invalid type and price column mappings

File: lusidtools/cocoon/utilities.py (hunk 3 above)
Comment (thomasdotred) [reply lusidtools/cocoon/utilities.py]: - changed to __adjusted_quote
- the `__adjusted_quote` column is created only a few lines before, why is it necessary to check this only a few lines later?
